android/src/main/java/io/customer/reactnative/sdk/extension/TypeConversion.kt

```diff
@@ -20,3 +25,24 @@ internal fun Double?.toCIOLogLevel(fallback: CioLogLevel = CioLogLevel.NONE): Ci
     return if (this == null) fallback
     else CioLogLevel.values().getOrNull(index = toInt() - 1) ?: fallback
 }
+
+/**
+ * Extensions function to build FCM [RemoteMessage] using RN map. This should be independent from
+ * the sender source and should be able to build a valid [RemoteMessage] for our native SDK.
+ *
+ * @param destination receiver of the message. It is mainly required for sending upstream messages,
+ * since we are using RemoteMessage only for broadcasting messages locally, we can use any non-empty
+ * string for it.
+ */
+internal fun ReadableMap.toFCMRemoteMessage(destination: String): RemoteMessage {
+    return with(RemoteMessage.Builder(destination)) {
+        getMap("data")?.toMutableStringMap()?.let { data -> setData(data) }
```

Comment:
I think we should probably check `getMap("notification")` as well if it exists, because what if the payload is divided into multiple maps?

Reply:
Thanks for pointing this out. RNFirebase do exposes `notification` as well, but I can't really figure out how should we pass it to [`RemoteMessage`](https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/RemoteMessage) using [`RemoteMessage.Builder`](https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/RemoteMessage.Builder). Apparently, [RNFirebase creates it in similar fashion](https://github.com/invertase/react-native-firebase/blob/main/packages/messaging/android/src/main/java/io/invertase/firebase/messaging/ReactNativeFirebaseMessagingSerializer.java#L190-L220). I'm still taking a look if we can do this in a better way, but I'm stuck at it now and we might have to continue with this approach as this seems the only proper way of creating `RemoteMessage` locally. However, if someone has more ideas on how to improve this, please let me know.

Reply:
We can probably go down the route of merging maps of both notification and data and utilizing it maybe?

Reply:
🤔 Yeah, that's one way of keeping `notification` params. Since this `RemoteMessage` is to be used by our SDK only, I don't see harm in this. I'll try this out and let you know.

Reply:
Updated code to merge these objects 🥪